File: aoc2024/java/src/test/java/com/peeekay/aoc2024/java/Day06Test.java
package com.peeekay.aoc2024.java;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class Day06Test {
    final Day06 day = new Day06(true);

    @Test
    void part1() {
        Assertions.assertEquals(41L, day.part1());
    }

    @Test
    void part2() {
        Assertions.assertEquals(6L, day.part2());
    }
}

File: test/com/peeekay/aoc2022/kotlin/Day06Test.kt
package com.peeekay.aoc2022.kotlin

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

@DisplayName("Day 06")
internal class Day06Test {

    @Test
    fun `Part1 with test data`() {
        val day = Day06(isTest = true)
        val answer = day.partOne()

        assertEquals(7, answer)
    }

    @Test
    fun `Part2 with test data`() {
        val day = Day06(isTest = true)
        val answer = day.partTwo()

        assertEquals(19, answer)
    }
}